quickstart/backend/src/main/java/com/digitalasset/quickstart/service/AppInstallRequestsApiImpl.java

```diff
@@ -42,128 +51,199 @@ public AppInstallRequestsApiImpl(
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-        });
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        methodSpan.addEvent("Starting acceptAppInstallRequest");
+        methodSpan.setAttribute("contractId", contractId);
+        methodSpan.setAttribute("commandId", commandId);
+        logger.info("acceptAppInstallRequest: contractId='{}', commandId='{}'", contractId, commandId);
+
+        // Capture the provider party asynchronously
+        return authenticatedPartyService.getPartyOrFail()
+                .thenCompose(providerParty ->
+                        damlRepository.findAppInstallRequestById(contractId)
+                                .thenCompose(contract -> {
+                                    methodSpan.addEvent("Fetched contract, exercising AppInstallRequest_Accept choice");
+
+                                    quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Accept choice =
+                                            new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Accept(
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestAccept.getInstallMeta().getData()),
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestAccept.getMeta().getData())
+                                            );
+
+                                    return ledger.exerciseAndGetResult(providerParty, contract.contractId, choice, commandId)
+                                            .thenApply(appInstallContractId -> {
+                                                methodSpan.addEvent("Choice exercised, building response AppInstall");
+                                                AppInstall appInstall = new AppInstall();
+                                                appInstall.setDso(contract.payload.getDso.getParty);
+                                                appInstall.setProvider(contract.payload.getProvider.getParty);
+                                                appInstall.setUser(contract.payload.getUser.getParty);
+                                                appInstall.setMeta(appInstallRequestAccept.getInstallMeta());
+                                                appInstall.setNumLicensesCreated(0);
+                                                return ResponseEntity.ok(appInstall);
+                                            });
+                                })
+                )
+                .whenComplete(
+                        completeWithin(parentContext, (res, ex) -> {
+                            if (ex == null) {
+                                logger.info("acceptAppInstallRequest: Success for contractId='{}'", contractId);
```

Comment:
This is not a particularly useful log, so should probably be at DEBUG. The INFO log for this should be up on line 83/4 where we have the useful information that needs to be included in this log.